datafusion/optimizer/src/rewrite_dependent_join.rs

```diff
@@ -693,6 +730,21 @@ impl TreeNodeRewriter for DependentJoinRewriter {
                         TreeNodeRecursion::Jump,
                     ));
                 }
+
+                // If expr has correlated subquery.
+                // TODO: what if both child and expr has subquery?
+                if let Some(filter) = &join.filter {
```

Comment:
:+1: i missed this case, join condition can contains subquery

Reply:
i can help resolve this TODO later, i think the decorrelation logic will fail in the case described above